solarforecastarbiter/metrics/calculator.py

```diff
@@ -149,6 +149,13 @@ def calculate_deterministic_metrics(processed_fx_obs, categories, metrics,
     elif len(metrics) == 0:
         raise RuntimeError("No metrics specified.")
 
+    # Force `groupby` to be consistent with `interval_label`, i.e., if
+    # `interval_label == ending`, then the last interval should be in the bin
+    closed = datamodel.CLOSED_MAPPING[processed_fx_obs.interval_label]
+    if closed == "ending":
+        obs.index -= pd.Timedelta("1ns")
+        fx.index -= pd.Timedelta("1ns")
```

Comment:
Should we be creating new series instead of modifying in place? My understanding is that these lines will modify the attributes of the `processed_fx_obs` which is probably not desirable.

Reply:
I'm not 100% sure, but to be safe, I'll add a `copy` operation and apply the 1ns shift to the copies.